### ISSUE TITLE
e2e tests: dynamically fetch the image digests

### ISSUE THREAD

**Issue by listx** [kubernetes-sigs/k8s-container-image-promoter#119]
Instead of hardcoding the digests in the `test-e2e` directory, dynamically calculate it on-the-fly, as a human would, for authoring the manifests and promoting images with them.

This is necessary because the image digests for the golden images depend on the gzipping mechanism that lies on the client machine (not server-side). So technically it is possible that the gzip library on the developer workstation (where the tests originated) differ from the gzip library where the e2e tests are executed later on, resulting in completely different digests (which would make every new PR fail).

This isn't a huge deal but it is a known issue.

The solution would be to fetch/calculate the image digests for all golden images used in the e2e tests after they are pushed up, and then to populate the yaml content with these values, before running them. Some amount of templating might be involved here as well.

TL;DR: remove hardcoded values like `2af5205553239e0eb17c544bc1e90c2d6173012a191f0a44131aa2f9c44bb511` from the e2e tests and instead generate them dynamically.

**Comment by jonjohnsonjr** [comment]
> This is necessary because the image digests for the golden images depend on the gzipping mechanism that lies in the registry (server-side).

This is not true. Clients gzip layers before uploading them and ungzip them after downloading. Did you see this somewhere?

**Comment by listx** [comment]
Oops, looks like I misremembered (your comment jogged my memory, and I remember it being client-side from an email thread many months ago). I still think this is an issue because the e2e tests are written on developer workstations and it could very well be that the gzip library on a developer machine doesn't match the gzip library used inside the Prow job. Will update the issue description.
